packages-web/maps/src/components/GoogleMap.tsx

```diff
@@ -0,0 +1,194 @@
+import classNames from "classnames";
+
+import googleApiWrapper from "./GoogleApi";
+import { Alert } from "@widgets-resources/piw-utils";
+import { getDimensions } from "../utils";
+import { Marker, SharedProps } from "../../typings";
+
+export interface GoogleMapsProps extends SharedProps {
+    scriptsLoaded?: boolean;
+    mapStyles?: string;
+
+    optionStreetView: boolean;
+    mapTypeControl: boolean;
+    fullScreenControl: boolean;
+    rotateControl: boolean;
+}
+
+export const GoogleMap = (props: GoogleMapsProps): ReactElement => {
+    const map = useRef<google.maps.Map>();
+    const googleMapsRef = useRef<HTMLDivElement>(null);
+    const defaultCenterLocation: google.maps.LatLngLiteral = { lat: 51.906688, lng: 4.48837 };
+    let bounds!: google.maps.LatLngBounds;
+
+    const markers = useRef<google.maps.Marker[]>([]); // Used to manage and remove markers from the map
+    const [validationMessage, setValidationMessage] = useState(props.validationMessage);
+
+    useEffect(() => {
+        if (props.scriptsLoaded) {
+            createMap();
+        }
+    }, []);
+
+    useEffect(() => {
+        addMarkers();
+    }, [props.locations, props.currentLocation]);
+
+    const createMap = (): void => {
+        const mapOptions: google.maps.MapOptions = {
+            zoom: props.zoomLevel,
+            zoomControl: props.optionZoomControl,
+            scrollwheel: props.optionScroll,
+            draggable: props.optionDrag,
+            streetViewControl: props.optionStreetView,
+            mapTypeControl: props.mapTypeControl,
+            fullscreenControl: props.fullScreenControl,
+            rotateControl: props.rotateControl,
+            minZoom: 1,
+            maxZoom: 20,
+            styles: getMapStyles()
+        };
+        if (googleMapsRef.current && !map.current) {
+            map.current = new google.maps.Map(googleMapsRef.current, mapOptions);
+            bounds = new google.maps.LatLngBounds().extend(defaultCenterLocation);
+        } else if (map.current) {
+            map.current.setOptions(mapOptions);
+        }
+        addMarkers();
+    };
+
+    const addMarkers = (): void => {
+        markers.current.forEach(marker => marker.setMap(null));
+        markers.current = [];
+        if (props.locations && props.locations.length > 0) {
+            bounds = new google.maps.LatLngBounds();
+            markers.current = props.locations.reduce<google.maps.Marker[]>((markerArray, currentLocation) => {
+                const marker = addMarker(currentLocation);
+                if (marker) {
+                    markerArray.push(marker);
+                }
+                return markerArray;
+            }, []);
+        }
+        addCurrentLocation();
+        updateCamera();
+    };
+
+    const addCurrentLocation = (): void => {
+        if (props.currentLocation) {
+            const currentLocationMarker = addMarker(props.currentLocation);
+            if (currentLocationMarker) {
```

Comment:
`currentLocationMarker && markers.current.push(currentLocationMarker);`

Reply:
ESLint: Expected an assignment or function call and instead saw an expression.(no-unused-expressions)